sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,344 @@
+    ==============      ==========================================
+
+    Parameters
+    ----------
+    subset : None, 'SA', 'SF', 'http', 'smtp'
+        To return the corresponding classical subsets of kddcup 99.
+        If None, return the entire kddcup 99 dataset.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        Random state for shuffling the dataset.
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    shuffle : bool, default=False
+        Whether to shuffle dataset.
+
+    percent10 : bool, default=False
+        Whether to load only 10 percent of the data.
+
+    Returns
+    -------
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
+        'data', the data to learn and 'target', the regression target for each
+        sample.
+    """
+    kddcup99 = fetch_kddcup99(shuffle=shuffle, percent10=percent10)
+
+    data = kddcup99.data
+    target = kddcup99.target
+
+    if subset == 'SA':
+        s = target == 'normal.'
+        t = np.logical_not(s)
+        normal_samples = data[s, :]
+        normal_targets = target[s]
+        abnormal_samples = data[t, :]
+        abnormal_targets = target[t]
+
+        n_samples_abnormal = abnormal_samples.shape[0]
+        # selected abnormal samples:
+        random_state = check_random_state(random_state)
+        r = random_state.randint(0, n_samples_abnormal, 3377)
+        abnormal_samples = abnormal_samples[r]
+        abnormal_targets = abnormal_targets[r]
+
+        data = np.r_[normal_samples, abnormal_samples]
+        target = np.r_[normal_targets, abnormal_targets]
+
+    if subset == 'SF' or subset == 'http' or subset == 'smtp':
+        # select all samples with positive logged_in attribute:
+        s = data[:, 11] == 1
+        data = np.c_[data[s, :11], data[s, 12:]]
+        target = target[s]
+
+        data[:, 0] = np.log((data[:, 0] + 0.1).astype(float))
+        data[:, 4] = np.log((data[:, 4] + 0.1).astype(float))
+        data[:, 5] = np.log((data[:, 5] + 0.1).astype(float))
+
+        if subset == 'http':
+            s = data[:, 2] == 'http'
+            data = data[s]
+            target = target[s]
+            data = np.c_[data[:, 0], data[:, 4], data[:, 5]]
+
+        if subset == 'smtp':
+            s = data[:, 2] == 'smtp'
+            data = data[s]
+            target = target[s]
+            data = np.c_[data[:, 0], data[:, 4], data[:, 5]]
+
+        if subset == 'SF':
+            data = np.c_[data[:, 0], data[:, 2], data[:, 4], data[:, 5]]
+
+    return Bunch(data=data, target=target)
+
+
+def fetch_kddcup99(subset=None, data_home=None, download_if_missing=True,
+                   random_state=None, shuffle=False, percent10=False):
```

Comment:
`fetch_kddcup99` loads the brute data, and `load_kddcup99` creates from these data the classical sets SA, SF, http and smtp. If there is no need to be able to get brute data, I can include fetch_kddcup99 in load_kddcup99.